web/html/src/manager/virtualization/nets/network-properties.js

```diff
@@ -0,0 +1,432 @@
+// @flow
+
+import type { ActionChain } from 'components/action-schedule';
+
+import * as React from 'react';
+import Validation from 'components/validation';
+import { Loading } from 'components/utils/Loading';
+import { Form } from 'components/input/Form';
+import { Radio } from 'components/input/Radio';
+import { Panel } from 'components/panels/Panel';
+import { Text } from 'components/input/Text';
+import { Range } from 'components/input/Range';
+import { Select } from 'components/input/Select';
+import { Check } from 'components/input/Check';
+import { unflattenModel, flattenModel } from 'components/input/form-utils';
+import { SubmitButton, Button } from 'components/buttons';
+import { Messages } from 'components/messages';
+import { ActionSchedule } from 'components/action-schedule';
+import { Utils } from 'utils/functions';
+import { VirtualizationNetworkDevsApi } from './virtualization-network-devs-api';
+import { VirtualPortFields } from './properties/VirtualPortFields';
+import { Interface } from './properties/Interface';
+import { Vlans } from './properties/Vlans';
+import { IpConfig } from './properties/IpConfig';
+import { DnsConfig } from './properties/DnsConfig';
+import * as FieldsData from './properties/fields-data';
+import * as utils from './properties/utils';
+
+import type {MessageType} from 'components/messages';
+
+type Props = {
+  serverId: string,
+  submitText: string,
+  submit: (Object) => void,
+  initialModel: ?Object,
+  messages: Array<MessageType>,
+  localTime: string,
+  timezone: string,
+  actionChains: Array<ActionChain>,
+};
+
+function clearFields(initialModel, setModel) {
+  if (initialModel) {
+    // Flatten the model for the form
+    let flattened = flattenModel(initialModel);
+    setModel(flattened);
+  } else {
+    setModel({});
+  }
+};
+
+export function NetworkProperties(props: Props) {
+  const [model, setModel] = React.useState(props.initialModel ? flattenModel(props.initialModel) : {});
+  const [invalid, setInvalid] = React.useState(false);
+  const [actionChain, setActionChain] = React.useState(null);
+  const [earliest, setEarliest] = React.useState(Utils.dateWithTimezone(props.localTime));
+
+  React.useEffect(() => {
+    clearFields(props.initialModel, setModel);
+  }, [props.initialModel]);
+
+  const onValidate = (isValid: boolean) => {
+    setInvalid(!isValid);
+  };
+
+  const onChange = (newModel: Object) => {
+    setModel(Object.assign({}, newModel));
+  };
+
+  const onSubmit = () => {
+    // Prepare the model for the submit action
+    let definition = unflattenModel(model, true);
+
+    const toClean = ['interface-selection-type', 'ipv6-enabled', 'virtualport-params-type']
+    toClean.forEach(field => {
+      if (definition[field] != null) {
+        delete definition[field];
+      }
+    });
+
+    // Rename the ipv4def and ipv6def properties
+    ["ipv4", "ipv6"].forEach(prop => {
```

Comment:
Perhaps I don't understand the intent here, why can't these fields have the intended name right away?

Reply:
the `unflattenModel()` would think those should be part of an array due to the number at the end of the name. To work this around at no code, I just ensured the name doesn't end with the number before that call and set the names right before sending the request. Not super elegant for sure, but avoids complex changes in the `unflattenModel()` function